### PR TITLE
risc-v/mpfs: Add error handling for PMP conf

### PR DESCRIPTION
**Pull request by jpaali**
Check return value of mpfs_board_pmp_setup() and
jump to mpfs_board_pmp_error() in case of error.

### CODE REVIEW

File: arch/risc-v/src/mpfs/mpfs_entrypoints.c [hunk 1]
```diff
@@ -107,6 +107,7 @@ void mpfs_jump_to_app(void)
       "la   sp, g_mpfs_boot_stacks\n"        /* Stack area base */
       "add  sp, sp, t0\n"                    /* Set stack pointer */
       "call mpfs_board_pmp_setup\n"          /* Run PMP configuration */
+      "bne  x10, x0, mpfs_board_pmp_error\n" /* If ret != 0, jump to errhan */
```
Comment (pussuw) [inline arch/risc-v/src/mpfs/mpfs_entrypoints.c:110]: x10 -> a0

**Review by pussuw — APPROVED** [review]
LGTM!

File: arch/risc-v/src/mpfs/mpfs_entrypoints.c (hunk 1 above)
Comment (jpaali) [reply arch/risc-v/src/mpfs/mpfs_entrypoints.c]: done

**Comment by jpaali** [comment]
This is now also in upstream.

**Pull request merged**